app/scripts/metamask-controller.js

```diff
@@ -2082,6 +2082,15 @@ export default class MetamaskController extends EventEmitter {
     this.controllerMessenger.subscribe(
       `${this.snapController.name}:snapRemoved`,
       (truncatedSnap) => {
+        this.metaMetricsController.trackEvent({
```

Comment:
I would suggest to place this part of code last in the process of this event, in case there is something wrong with the trackEvent/metrics and it throws an error or else. In my opinion metrics should be the last in execution (priority) order in the most cases.

Reply:
Fine by me!